server/channels/store/sqlstore/autotranslation_store.go

```diff
@@ -0,0 +1,443 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+
+	sq "github.com/mattermost/squirrel"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+)
+
+type SqlAutoTranslationStore struct {
+	*SqlStore
+}
+
+type TranslationMeta json.RawMessage
+
+type Translation struct {
+	ObjectType        string          `db:"object_type"`
+	ObjectID          string          `db:"object_id"`
+	DstLang           string          `db:"dst_lang"`
+	ProviderID        string          `db:"provider_id"`
+	NormHash          string          `db:"norm_hash"`
+	Text              string          `db:"text"`
+	Confidence        *float64        `db:"confidence"`
+	Meta              TranslationMeta `db:"meta"`
+	UpdateAt          int64           `db:"updateat"`
+	ContentSearchText *string         `db:"content_search_text"`
+}
+
+func (m *TranslationMeta) ToMap() (map[string]any, error) {
+	if m == nil {
+		return nil, nil
+	}
+	var result map[string]any
+	if err := json.Unmarshal(*m, &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func newSqlAutoTranslationStore(sqlStore *SqlStore) store.AutoTranslationStore {
+	return &SqlAutoTranslationStore{
+		SqlStore: sqlStore,
+	}
+}
+
+func (s *SqlAutoTranslationStore) IsChannelEnabled(channelID string) (bool, *model.AppError) {
+	query := s.getQueryBuilder().
+		Select("COALESCE((props->'autotranslation')::boolean, false)").
+		From("channels").
+		Where(sq.Eq{"id": channelID})
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return false, model.NewAppError("SqlAutoTranslationStore.IsChannelEnabled",
+			"store.sql_autotranslation.query_build_error", nil, err.Error(), 500)
+	}
+
+	var enabled *bool
+	if err := s.GetReplica().Get(&enabled, queryString, args...); err != nil {
+		if err == sql.ErrNoRows {
+			return false, model.NewAppError("SqlAutoTranslationStore.IsChannelEnabled",
+				"store.sql_autotranslation.channel_not_found", nil, "channel_id="+channelID, 404)
+		}
+		return false, model.NewAppError("SqlAutoTranslationStore.IsChannelEnabled",
+			"store.sql_autotranslation.get_channel_enabled.app_error", nil, err.Error(), 500)
+	}
+
+	if enabled == nil {
+		return false, nil
+	}
+
+	return *enabled, nil
+}
+
+func (s *SqlAutoTranslationStore) SetChannelEnabled(channelID string, enabled bool) *model.AppError {
+	query := s.getQueryBuilder().
+		Update("channels").
+		Set("props", sq.Expr(
+			"jsonb_set(COALESCE(props, '{}'::jsonb), '{autotranslation}', (?::jsonb), true)",
```

Comment:
We don't need the first COALESCE, right? The column is defined as defaulting to `'{}'::jsonb`. Also, it seems that the default for the last argument is `true`, so we don't really need it here, and I think it would help read this:
```suggestion
			"jsonb_set(props, '{autotranslation}', (?::jsonb))",
```